### PR TITLE
refactor: stencil java client

### PR DESCRIPTION
**Pull request by rohilsurana**
**Refactoring**
- [x] **BREAKING** Rename package group from `com.gojek.de` to `io.odpf`
- [x] **BREAKING** Update `StencilConfig` to use ~~setters~~ builder pattern instead of parsing from `Map<string, string>`
- [x] ~~Expose refresh jitter as a config~~ Tracked in separate issue #21 
- [x] Add config for setting http bearer auth token for stencil client
- [x] Push jar to Github packages instead of Sonatype
- [x] Integrate Github workflows for publishing jar
- [x] Add java doc comments for all public facing APIs
- [x] Update `README.md` with usage

### CODE REVIEW

File: clients/java/src/main/java/io/odpf/stencil/client/URLStencilClient.java [hunk 1]
```diff
@@ -38,17 +38,17 @@ public URLStencilClient(String url, StencilConfig config, DescriptorCacheLoader
     }
 
     public URLStencilClient(String url, StencilConfig stencilConfig, DescriptorCacheLoader cacheLoader, Ticker ticker) {
-        this.shouldAutoRefreshCache = stencilConfig.shouldAutoRefreshCache();
-        this.ttl = stencilConfig.getTilInMinutes() != 0 ? Duration.ofMinutes(stencilConfig.getTilInMinutes()) :
+        this.shouldAutoRefreshCache = stencilConfig.getCacheAutoRefresh();
+        this.ttl = stencilConfig.getCacheTtlMs() != 0 ? Duration.ofMillis(stencilConfig.getCacheTtlMs()) :
```
Comment (sravankorumilli) [inline clients/java/src/main/java/io/odpf/stencil/client/URLStencilClient.java:56]: The conversion to Duration, I find it unnecessary, as we are using long everywhere

File: clients/java/build.gradle [hunk 2]
```diff
@@ -101,25 +92,8 @@ publishing {
                         email = 'de@go-jek.com'
```
Comment (sravankorumilli) [inline clients/java/build.gradle:91]: this email can be removed, check whether its mandatory

File: clients/java/build.gradle [hunk 3]
```diff
@@ -3,12 +3,13 @@ plugins {
...
-    id "com.google.protobuf" version "0.8.15"
-    id "io.freefair.lombok" version "5.3.3.3"
+    id 'com.google.protobuf' version '0.8.15'
+    id 'io.freefair.lombok' version '5.3.3.3'
+    id 'com.palantir.git-version' version '0.12.3'
 }
 
 group 'io.odpf'
-version '5.0.0'
+version gitVersion()
```
Comment (sravankorumilli) [inline clients/java/build.gradle]: We will be following semversion with vM.m.m convention, if that is true we will be having problems with this

Comment (sravankorumilli) [reply clients/java/build.gradle]: I believe you have fixed this already

File: clients/java/lombok.config [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# This file is generated by the 'io.freefair.lombok' Gradle plugin
+config.stopBubbling = true
```
Comment (sravankorumilli) [inline clients/java/lombok.config:2]: Is this file needed to be committed what is the purpose of this?

File: clients/java/build.gradle (hunk 3 above)
Comment (rohilsurana) [reply clients/java/build.gradle]: yepp this was fixed

File: clients/java/src/main/java/io/odpf/stencil/http/RetryHttpClient.java [hunk 5]
```diff
@@ -26,6 +28,12 @@ public CloseableHttpClient create(StencilConfig stencilConfig) {
         long backoffMs = stencilConfig.getFetchBackoffMinMs() != 0 ? stencilConfig.getFetchBackoffMinMs() :
                 new RandomUtils().getRandomNumberInRange(DEFAULT_STENCIL_BACKOFF_MS_MIN, DEFAULT_STENCIL_BACKOFF_MS_MAX);
         int retries = stencilConfig.getFetchRetries();
+        List<Header> defaultHeaders = new ArrayList<>();
+
+        if (stencilConfig.getFetchAuthBearerToken() != null) {
```
[the remark this answers is not in the document]
Comment (rohilsurana) [reply clients/java/src/main/java/io/odpf/stencil/http/RetryHttpClient.java:34]: I was a bit b/w both of them and actually the thing is by default the value of this config is null. So if and only if the user has set this value, will this have a non null value. 
So basically if the developer using the library wants to set empty this will allow him to do it, instead of magically not setting it. wdyt?

[the remark this answers is not in the document]
Comment (sravankorumilli) [reply clients/java/src/main/java/io/odpf/stencil/http/RetryHttpClient.java:34]: sure, even empty check doesn't change anything
